pkg/influx/auth_test.go

```diff
@@ -0,0 +1,105 @@
+package influx
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/influx2cortex/pkg/errorx"
+	"github.com/grafana/influx2cortex/pkg/remotewrite"
+	"github.com/grafana/influx2cortex/pkg/remotewrite/remotewritemock"
+	"github.com/grafana/influx2cortex/pkg/server"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+)
+
+func TestAuthentication(t *testing.T) {
+	tests := []struct {
+		name         string
+		url          string
+		data         string
+		enableAuth   bool
+		orgID        string
+		expectedCode int
+		expectedErr  error
+	}{
+		{
+			name:         "test auth enabled valid org ID",
+			url:          "/write",
+			data:         "measurement,t1=v1 f1=2 1465839830100400200",
+			enableAuth:   true,
+			orgID:        "valid",
+			expectedCode: http.StatusNoContent,
+			expectedErr:  nil,
+		},
+		{
+			name:         "test auth enabled invalid org ID",
+			url:          "/write",
+			data:         "measurement,t1=v1 f1=2 1465839830100400200",
+			enableAuth:   true,
+			orgID:        "",
+			expectedCode: http.StatusUnauthorized,
+			expectedErr:  errorx.BadRequest{},
+		},
+		{
+			name:         "test auth disabled",
+			url:          "/write",
+			data:         "measurement,t1=v1 f1=2 1465839830100400200",
+			enableAuth:   false,
+			orgID:        "fake",
```

Comment:
one more case to detect -- no orgid with auth off is also a success

Reply:
Yeah, it does test this because in auth is enabled, the test doesn't inject the org ID into the request, and relies on the middleware.HttpFakeAuth{} to inject the "fake" org ID into the request's context. This orgID for this test is just used to verify the "fake" orgID is added to the request context. But if there is a way to make this more clear, let me know!